lua/entities/gmod_tardis_debug_pointer/shared.lua

```diff
@@ -80,4 +80,365 @@ concommand.Add("tardis2_debug_pointer", function(ply,cmd,args)
+
+        local function SetupProperty(category, name, value, a, b, c, d)
+            local vmin, vmax, update_func
+            local vtype = "Float"
+            if isnumber(b) then
+                vmin = a
+                vmax = b
+                update_func = c
+            else
+                vmin = value - a
+                vmax = value + a
+                update_func = b
+            end
+
+            if d then
+                vtype = d
+            end
+
+            local row1 = pr:CreateRow( category, name )
+            local row2 = pr:CreateRow( category, name .. " (precise)" )
+
+            row1:Setup( vtype, { min = vmin, max = vmax } )
+            row1:SetValue(value)
+            row1.DataChanged = function( _, val )
+                row2:SetValue(val)
+                update_func(val)
+            end
+
+            row2:Setup( "Generic" )
+            row2:SetValue(value)
+            row2.DataChanged = function( _, val )
+                if tonumber(val) == nil then return end
+                row1:SetValue(val)
+                update_func(val)
+            end
+
+            return row1, row2
+        end
+
+        x, x2 = SetupProperty("Position", "X", px, 100, function(val)
+            px = val
+            UpdatePointerPos()
+        end)
+        y, y2 = SetupProperty("Position", "Y", py, 100, function(val)
+            py = val
+            UpdatePointerPos()
+        end)
+        z, z2 = SetupProperty("Position", "Z", pz, 100, function(val)
+            pz = val
+            UpdatePointerPos()
+        end)
+
+        xr, xr2 = SetupProperty("Relative position", "Forward / Back", prx, 100, function(val)
+            prx = val
+            UpdatePointerPos(true)
+        end)
+        yr, yr2 = SetupProperty("Relative position", "Right / Left", pry, 100, function(val)
+            pry = val
+            UpdatePointerPos(true)
+        end)
+        zr, zr2 = SetupProperty("Relative position", "Up / Down", prz, 100, function(val)
+            prz = val
+            UpdatePointerPos(true)
+        end)
+
+        xpv, xpv2 = SetupProperty("Player view position", "Forward / Back", ppvx, 400, function(val)
+            ppvx = val
+            UpdatePointerPos(false, true)
+        end)
+        ypv, ypv2 = SetupProperty("Player view position", "Right / Left", ppvy, 400, function(val)
+            ppvy = val
+            UpdatePointerPos(false, true)
+        end)
+        zpv, zpv2 = SetupProperty("Player view position", "Up / Down", ppvz, 400, function(val)
+            ppvz = val
+            UpdatePointerPos(false, true)
+        end)
+
+        local pv_iva = pr:CreateRow( "Player view position", "Ignore vertical angle" )
+        pv_iva:Setup( "Bool" )
+        pv_iva:SetValue(pv_ignore_vertical_angle)
+        pv_iva.DataChanged = function( _, val )
+            pv_ignore_vertical_angle = (val == 1)
+        end
+
+        local ap, ap2 = SetupProperty( "Angle", "Pitch", ang_p, 360, function(val)
```

Comment:
This seems to glitch out and keep flipping 180 degrees

Reply:
I've added this one to #987
Hope it's okay to leave it since it's a developer tool, not a TARDIS feature

Reply:
Yeah that's fine just wanted to note it